experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/lazyField.spec.ts

```diff
@@ -65,3 +92,316 @@ describe("lazyField", () => {
+			assert(!field.is(SchemaBuilder.fieldOptional(booleanLeafSchema)));
+			assert(!field.is(SchemaBuilder.fieldValue(Any)));
+			assert(!field.is(SchemaBuilder.fieldSequence(Any)));
+			assert(!field.is(SchemaBuilder.fieldRecursive(FieldKinds.optional, booleanLeafSchema)));
+		});
+
+		it("Boolean", () => {
+			const builder = new SchemaBuilder("test");
+			const booleanLeafSchema = builder.leaf("bool", ValueSchema.Boolean);
+			const numberLeafSchema = builder.leaf("number", ValueSchema.Number);
+
+			const { context, cursor } = createSingleValueTree(
+				builder,
+				SchemaBuilder.fieldOptional(builder.struct("struct", {})),
+				{},
+			);
+
+			assert.equal(
+				context.forest.tryMoveCursorToField(detachedFieldAnchor, cursor),
+				TreeNavigationResult.Ok,
+			);
+
+			const field = new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(booleanLeafSchema),
+				cursor,
+				detachedFieldAnchor,
+			);
+
+			// Positive cases
+			assert(field.is(SchemaBuilder.fieldOptional(booleanLeafSchema)));
+			assert(field.is(SchemaBuilder.fieldRecursive(FieldKinds.optional, booleanLeafSchema)));
+
+			// Negative cases
+			assert.equal(field.is(SchemaBuilder.fieldValue(Any)), false);
+			assert.equal(field.is(SchemaBuilder.fieldValue(booleanLeafSchema)), false);
+			assert.equal(field.is(SchemaBuilder.fieldValue(numberLeafSchema)), false);
+			assert.equal(field.is(SchemaBuilder.fieldSequence(Any)), false);
+			assert.equal(field.is(SchemaBuilder.fieldSequence(booleanLeafSchema)), false);
+			assert.equal(field.is(SchemaBuilder.fieldSequence(numberLeafSchema)), false);
+			// assert.equal(
+			// 	field.is(SchemaBuilder.fieldRecursive(FieldKinds.optional, numberLeafSchema)),
+			// 	false,
+			// );
+		});
+
+		// TODO: what other cases are interesting?
+	});
+
+	describe("length", () => {
+		it("No value", () => {
+			const builder = new SchemaBuilder("test");
+			const numberLeafSchema = builder.leaf("number", ValueSchema.Number);
+			const rootSchema = SchemaBuilder.fieldOptional(numberLeafSchema);
+
+			const { context, cursor } = createSingleValueTree(builder, rootSchema, undefined);
+
+			const field = new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(Any),
+				cursor,
+				rootFieldAnchor,
+			);
+
+			assert.equal(field.length, 0);
+		});
+
+		it("With value", () => {
+			const builder = new SchemaBuilder("test");
+			const numberLeafSchema = builder.leaf("number", ValueSchema.Number);
+			const rootSchema = SchemaBuilder.fieldOptional(numberLeafSchema);
+
+			const { context, cursor } = createSingleValueTree(builder, rootSchema, 42);
+
+			const field = new LazyOptionalField(
+				context,
+				SchemaBuilder.fieldOptional(numberLeafSchema),
+				cursor,
+				rootFieldAnchor,
+			);
+
+			assert.equal(field.length, 1);
+		});
+	});
+
+	/**
+	 * Creates a single-node, primitive tree, and returns a field associated with that node.
+	 */
+	function createPrimitiveField(
```

Comment:
in our codebase, when we use "primitive", we are referring to JS primitive types (immutable, value semantic types which can be compared with ===). We have Value, TreeValue, and leaf as concepts which include support for the common primitive types as well as fluid handles, but fluid handles are not primitives, so if you want to support them here your naming is wrong.

Reply:
also as you are testing fields, details of leaf nodes don't really belong here: that sounds like something for lazyTree.spec.ts

Reply:
> also as you are testing fields, details of leaf nodes don't really belong here: that sounds like something for lazyTree.spec.ts

I'm strictly using the leaf node as an implementation detail to construct a tree with a value node to associate the test field with. Is there an easier / better way to accomplish this?